File: qlack-base-application-server/src/main/java/com/eurodyn/qlack/baseapplication/dto/UserDTO.java
package com.eurodyn.qlack.baseapplication.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.hibernate.validator.constraints.Length;
import org.springframework.data.annotation.ReadOnlyProperty;

@Data
@NoArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class UserDTO extends BaseDTO {

  @Email
  String email;
  @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[\\W\\_])[A-Za-z\\d\\W\\_]{8,}$")
  String password;
  @Length(min = 10, max = 60)
  String firstname;
  @Length(min = 10, max = 60)
  String lastname;
  @ReadOnlyProperty
  @NotNull
  byte status;
}

File: qlack-base-application-ui/src/app/auth/dto/user-dto.ts
import {BaseDto} from "../../dto/base-dto";
import {FileDto} from "../../files/dto/file-dto";
import {ExtraInfoDto} from "../../dto/extra-info-dto";

export class UserDto extends BaseDto {
  email?: string;
  password?: string;
  firstname?: string;
  lastname?: string;
  status?: number;
  profilepic?: FileDto;
  extraInfo?: ExtraInfoDto;
}
